be/src/vec/functions/function_convert_tz.h

```diff
@@ -161,13 +180,16 @@ class FunctionConvertTZ : public IFunction {
                                assert_cast<const ColumnString*>(argument_columns[2].get()),
                                assert_cast<ColumnDateTimeV2*>(result_column.get()),
                                assert_cast<ColumnUInt8*>(result_null_map_column.get())->get_data(),
-                               input_rows_count);
+                               input_rows_count, _time_zone_cache);
             block.get_by_position(result).column = ColumnNullable::create(
                     std::move(result_column), std::move(result_null_map_column));
         }
 
         return Status::OK();
     }
+
+private:
+    std::map<StringRef, cctz::time_zone> _time_zone_cache;
```

Comment:
Is this thread-safe?

Reply:
No. But the map is a separate for each thread

Reply:
After checking again, the function was not separate for each thread.
I will put the cache map into function state of FunctionContext.
Thanks.